python/pyngraph/node_factory.cpp

```diff
@@ -0,0 +1,177 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <pybind11/stl.h>
+
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/check.hpp"
+#include "ngraph/except.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/util.hpp"
+#include "node_factory.hpp"
+
+class DictAttributeDeserializer : public ngraph::AttributeVisitor
+{
+public:
+    DictAttributeDeserializer(const py::dict& attributes)
+        : m_attributes(attributes)
+    {
+    }
+
+    virtual void on_attribute(const std::string& name, std::string& value) override
+    {
+        if (m_attributes.contains(name))
+        {
+            value = m_attributes[name.c_str()].cast<std::string>();
+        }
+    }
+    virtual void on_attribute(const std::string& name, bool& value) override
+    {
+        if (m_attributes.contains(name))
```

Comment:
Why this if is needed every where?
What happen when name is absent from `m_attributes`?

Reply:
We have to check whether the requested attribute name is present in the dictionary we received. If the attribute name is absent we just don't update it.